src/transformers/quantizers/quantizer_bnb_4bit.py

```diff
@@ -171,6 +172,9 @@ def create_quantized_param(
 
         old_value = getattr(module, tensor_name)
 
+        # `torch.Tensor.to(<int num>)` is not supported by `torch_npu` (see this [issue](https://github.com/Ascend/pytorch/issues/16)).
+        if isinstance(target_device, int) and is_torch_npu_available():
+            target_device = f"npu:{target_device}"
```

Comment:
Apart from 8bit not being implemented on the bnb side yet, is there a reason this would not also be needed in the `Bnb8BitHfQuantizer` as well?

Reply:
@matthewdouglas Thanks for your review. The change to `Bnb8BitHfQuantizer` was just to make it easier for us to verify that support for bnb 8bit was correct. Apparently it was unnecessary and I have removed it :-)